src/test/java/healthy/lifestyle/backend/admin/workout/service/ExerciseAdminServiceTest.java

```diff
@@ -0,0 +1,157 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+
+@ExtendWith(MockitoExtension.class)
+public class ExerciseAdminServiceTest {
+
+    @InjectMocks
+    ExerciseAdminServiceImpl exerciseAdminService;
+
+    @Mock
+    private ExerciseAdminRepository exerciseAdminRepository;
+
+    @Spy
+    private TestUtil testUtil;
+
+    @Spy
+    ModelMapper modelMapper;
+
+    @ParameterizedTest
+    @MethodSource("multipleFilters")
+    void getExercisesByFiltersTest_shouldReturnEntityResponseDtoList_whenParamsAreValid(
+            String title, String description, Boolean isCustom, Boolean needsEquipment, List<Integer> resultSeeds) {
+        // Given
+        User user1 = testUtil.createUser(1);
+        User user2 = testUtil.createUser(2);
+
+        BodyPart bodyPart1 = testUtil.createBodyPart(1);
+        BodyPart bodyPart2 = testUtil.createBodyPart(2);
+        BodyPart bodyPart3 = testUtil.createBodyPart(3);
+        BodyPart bodyPart4 = testUtil.createBodyPart(4);
+
+        HttpRef defaultHttpRef1 = testUtil.createDefaultHttpRef(1);
+        HttpRef defaultHttpRef2 = testUtil.createDefaultHttpRef(2);
+
+        HttpRef customHttpRef1 = testUtil.createCustomHttpRef(3, user1);
+        HttpRef customHttpRef2 = testUtil.createCustomHttpRef(4, user1);
+        HttpRef customHttpRef3 = testUtil.createCustomHttpRef(5, user2);
+        HttpRef customHttpRef4 = testUtil.createCustomHttpRef(6, user2);
+
+        Exercise defaultExercise1 =
+                testUtil.createDefaultExercise(1, true, List.of(bodyPart1), List.of(defaultHttpRef1));
+        Exercise defaultExercise2 =
+                testUtil.createDefaultExercise(2, false, List.of(bodyPart2), List.of(defaultHttpRef2));
+
+        Exercise customExercise1 = testUtil.createCustomExercise(
+                3, true, List.of(bodyPart3), List.of(customHttpRef1, customHttpRef2), user1);
+        Exercise customExercise2 = testUtil.createCustomExercise(
+                4, false, List.of(bodyPart4), List.of(customHttpRef3, customHttpRef4), user2);
+
+        List<Exercise> exercises = Arrays.asList(defaultExercise1, defaultExercise2, customExercise1, customExercise2);
+
+        List<Exercise> expectedExercises = filterExercises(exercises, title, description, isCustom, needsEquipment);
+
+        when(exerciseAdminRepository.findByFilters(eq(title), eq(description), eq(isCustom), eq(needsEquipment)))
+                .thenReturn(Optional.of(expectedExercises));
+
+        // When
+        List<ExerciseResponseDto> result =
+                exerciseAdminService.getExercisesByFilters(title, description, isCustom, needsEquipment);
+
+        // Then
+        Assertions.assertEquals(expectedExercises.size(), result.size());
```

Comment:
`Assertions.assertEquals(resultSeeds.size(), result.size());` serves the same goal since resultSeeds gives us an expected output size.